pkg/jtracer/jtracer.go

```diff
@@ -14,16 +14,77 @@
 
 package jtracer
 
-import "github.com/opentracing/opentracing-go"
+import (
+	"context"
+	"log"
+	"sync"
+
+	"github.com/opentracing/opentracing-go"
+	"go.opentelemetry.io/otel"
+	otbridge "go.opentelemetry.io/otel/bridge/opentracing"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+)
 
 type JTracer struct {
-	OT opentracing.Tracer
+	OT   opentracing.Tracer
+	OTEL trace.TracerProvider
 }
 
-func OT(t opentracing.Tracer) JTracer {
-	return JTracer{OT: t}
+var once sync.Once
+
+func New() JTracer {
+	opentracingTracer, otelTracerProvider := initBoth()
+	return JTracer{OT: opentracingTracer, OTEL: otelTracerProvider}
 }
 
 func NoOp() JTracer {
-	return JTracer{OT: opentracing.NoopTracer{}}
+	return JTracer{OT: opentracing.NoopTracer{}, OTEL: trace.NewNoopTracerProvider()}
+}
+
+// initBoth initializes OpenTelemetry SDK and uses OTel-OpenTracing Bridge
+func initBoth() (opentracing.Tracer, trace.TracerProvider) {
+	opts := []otlptracehttp.Option{otlptracehttp.WithInsecure()}
+	traceExporter, err := otlptrace.New(
+		context.Background(),
+		otlptracehttp.NewClient(opts...),
```

Comment:
it's better to use grpc, this is Jaeger's default protocol

Reply:
I suspect I may go wrong here.